types/react/index.d.ts

```diff
@@ -1223,6 +1223,7 @@ declare namespace React {
          * See the [DOM Level 3 Events spec](https://www.w3.org/TR/uievents-key/#named-key-attribute-values). for possible values
          */
         key: string;
+        isComposing: boolean;
```

Comment:
You're adding this property to React's synthetic KeyboardEvent which is not the same as the `KeyboardEvent` interface in the DOM (which you referred to).

React currently does not implement `isComposing` in their synthetic KeyboardEvent.

[Source code of React's synthetic KeyboardEvent](https://github.com/facebook/react/blob/6f8843837c4aec94d362560a021c3722d260f49f/packages/react-dom/src/events/SyntheticEvent.js#L423-L471)
[CodeSandbox illustrating that `isComposing` is `undefined`](https://codesandbox.io/s/react-eventiscomposing-96fdj?file=/src/App.js)

Reply:
Yeah, you're right.
I'll add this if Synthetic KeyboardEvent supports `isComposing`.
Thanks.